scripts/dataset/download_pl_additional_data.py

```diff
@@ -32,65 +30,48 @@ def main(
     batch_size: int = typer.Option(BATCH_SIZE),
     n_jobs: int = typer.Option(N_JOBS),
 ) -> None:
-    client: MongoClient[dict[str, Any]] = MongoClient(mongo_uri, server_api=ServerApi("1"))
-    collection = client["juddges"]["judgements"]
-    client.admin.command("ping")
-
     api = PolishCourtAPI()
 
     # find rows which are missing at least one field
     query = {"$or": [{field: {"$exists": False}} for field in api.schema[data_type.value]]}
 
+    collection = get_mongo_collection()
     num_docs_to_update = collection.count_documents(query)
     logger.info(f"There are {num_docs_to_update} documents to update")
 
     # fetch all ids at once to avoid cursor timeout
     cursor = collection.find(query, {"_id": 1}, batch_size=batch_size)
-    docs_to_update: list[str] = []
-    for doc in tqdm(cursor, total=num_docs_to_update, desc="Fetching doc list"):
-        docs_to_update.append(str(doc["_id"]))
-
-    batched_docs_to_update = (
-        docs_to_update[i : i + batch_size] for i in range(0, len(docs_to_update), batch_size)
-    )
-
-    download_update = DownloadDataAndUpdateDatabase(mongo_uri, data_type)
-    with WorkerPool(n_jobs=n_jobs) as pool:
-        pool.map_unordered(
-            download_update,
-            batched_docs_to_update,
-            progress_bar=True,
-            iterable_len=math.ceil(num_docs_to_update / batch_size),
+    batched_cursor = BatchedDatabaseCursor(cursor=cursor, batch_size=batch_size, prefetch=True)
+
+    download_data = DownloadAdditionalData(data_type)
+    download_data_and_update_db = BatchDatabaseUpdate(mongo_uri, download_data)
+
+    with multiprocessing.Pool(n_jobs) as pool:
+        list(
+            tqdm(
+                pool.imap_unordered(
+                    download_data_and_update_db,
+                    batched_cursor,
+                ),
+                total=math.ceil(num_docs_to_update / batch_size),
+            )
         )
 
+    assert collection.count_documents(query) == 0
+
 
-class DownloadDataAndUpdateDatabase:
-    def __init__(self, mongo_uri: str, data_type: DataType):
-        self.mongo_uri = mongo_uri
+class DownloadAdditionalData:
```

Comment:
```suggestion
class AdditionalDataDownloader:
```

verbs should be used for methods

Reply:
done